developer_guides/unit_tests.rst

```diff
@@ -80,10 +80,47 @@ Preparing cmocka package
 
 #. Use the target location for cmocka files when invoking *configure* script:
 
-   .. code-block:: bash
+.. code-block:: bash
 
       ./configure --with-cmocka-prefix=/home/<you>/cminstall ...
 
+Wrapping objects for unit tests
+******************************
+
+If you need to mock a symbol, define it in a unit test and include the .h file. 
+There are 2 cases where this isn't possible:
+
+*	Static functions in headers(those most probably are inline short functions
+	and don't have to be mocked)
+
+*	Static functions that are in the same file as tested functionality and are
+	exceedingly large so they can't be tested as one functionality. 
+
+Whatever the reason, mocking of those symbols can be done by using --wrap linker
+functionality. To wrap the symbol follow these steps:
+
+#. Create mocked symbol named __wrap_symbol_name
+
+#. Pass instruction for the linker -Wl, --wrap=symbol_name during compilation.
+
+Now every symbol call to symbol_name will call __wrap_symbol_name.
+
+Instructions can be passed to the linker in the SOF UT environment using
+CFLAGS, however they should be passed in separate variables in the makefile.
+
+Example:
+
+.. code-block:: bash
+ 
+	  # some tests before ...
+          check_PROGRAMS += pipeline_connect_upstream
+          pipeline_connect_upstream_SOURCES = ../../src/audio/pipeline.c src/audio/pipeline/ pipeline_mocks.c src/audio/pipeline/pipeline_connect_upstream.c src/audio/pipeline/pipeline_mocks_rzalloc.c

```

Comment:
better to break the long command lines and end with a continuation character ``\``